VstsTasks/AzureDtlCreateVM/task-funcs.ps1

```diff
@@ -417,15 +431,35 @@ function Wait-ApplyArtifacts
                 throw "Waited for more than $(ConvertTo-MinutesString $totalWaitMinutes). Failing the task."
             }
 
-            $continueWaiting = $true
-
             $vm = Get-AzureRmResource -ResourceId $ResourceId
-            if ($vm)
+            if (-not $vm)
             {
-                $provisioningState = $vm.Properties.provisioningState
-                $continueWaiting = $provisioningState -and ($provisioningState -eq 'ApplyingArtifacts' -or $provisioningState -eq 'UpgradingVmAgent')
+                throw "Unable to find VM with resource ID '$ResourceId'."
             }
 
+            $vmName = $vm.Name
+            if ($vm.ResourceName)
+            {
+                $vmLabName = $vm.ResourceName.Split('/')[0]
+                $vmFullName = $vm.ResourceName

```

Comment:
What is a VM *full* name?

Reply:
It contains the parent name. So, in the case of a Lab VM, it would be in the form `labName\vmName`. The resource name would only be `vmName`. But when we make the call to expand, we need the "full name". Otherwise, it doesn't work and it fails. Weird? Yes, but that's the change in the newer versions of the Azure PS cmdlet. 😞